pennylane/qnn/torch.py

```diff
@@ -447,21 +449,20 @@ def construct(self, args, kwargs):
         self.qnode.construct((), kwargs)
 
     def __getattr__(self, item):
-        """If the given attribute does not exist in the class, look for it in the wrapped QNode."""
+        """If the qnode is initialized, first check to see if the attribute is on the qnode."""
         if self._initialized:
-            return getattr(self.qnode, item)
+            with contextlib.suppress(AttributeError):
+                return getattr(self.qnode, item)
 
-        try:
-            return self.__dict__[item]
-        except KeyError as exc:
-            raise AttributeError(item) from exc
+        return super().__getattr__(item)
 
     def __setattr__(self, item, val):
-        """If the given attribute does not exist in the class, try to set it in the wrapped QNode."""
-        if self._initialized:
+        """If the qnode is initialized and item is already a qnode property, update it on the qnode, else
+        just update the torch layer itself."""
+        if self._initialized and item in self.qnode.__dict__:
```

Comment:
Why not do this same check in `__getattr__` rather than suppressing the `AttributeError`?

Reply:
It could also be an option, but I'm unsure if it matters in this case.